src/components/duration.rs

```diff
@@ -22,6 +23,40 @@ pub use date::DateDuration;
 #[doc(inline)]
 pub use time::TimeDuration;
 
+#[repr(i8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum DurationSign {
+    Positive = 1,
+    Zero = 0,
+    Negative = -1,
+}
+
+impl From<Ordering> for DurationSign {
+    fn from(value: Ordering) -> Self {
+        match value {
+            Ordering::Greater => Self::Positive,
+            Ordering::Equal => Self::Zero,
+            Ordering::Less => Self::Negative,
+        }
+    }
+}
```

Comment:
Hmmm, I don't know if it makes much sense to have a conversion from these enums. `Ordering` is a completely different concept than `DurationSign`, and things like `DurationSign::from(zero.cmp(&target))` don't make much sense if you think about it.

Reply:
Yeah, it's a bit awkward. I wasn't super in love with it. Maybe just a `From<i8>`